### PR TITLE
Integrate initializeJob with queueTask

### PR DESCRIPTION
**Pull request by yoninachmany**
r @sandlerben @anoshaminai 
When initiateTranscriptionJobHandler is reached, queue a task using the task executor

Task function in transcription combines transcription functions (except SendEmail) to create a task function for transcription in the web app.

screen shot 2016-03-18 at 1 09 34 pm

### CODE REVIEW

File: web/routes.go [hunk 1]
```diff
@@ -64,7 +65,9 @@ func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	fmt.Fprintf(w, "Accepted!")
+	id := tasks.DefaultTaskExecuter.QueueTask(transcription.Task(jsonData.AudioURL, jsonData.EmailAddresses))
```
Comment (sandlerben) [inline web/routes.go]: Change to

``` go
executer := tasks.DefaultTaskExecuter
id := executer.QueueTask(<>)
```

File: transcription/transcription.go [hunk 2]
```diff
@@ -76,3 +77,17 @@ func fileNameFromURL(url string) string {
 	fileName := tokens[len(tokens)-1]
 	return fileName
 }
+
+// Task returns a task function for transcription using transcription functions.
+func Task(audioURL string, emailAddresses []string) func() error {
```
Comment (sandlerben) [inline transcription/transcription.go]: Make sure to test this function. Also, are we testing `fileNameFromUrl` or `DownloadFileFromUrl`?

Comment (sandlerben) [inline transcription/transcription.go]: Rename this to something like `MakeTaskFunction`

File: web/routes.go (hunk 1 above)
Comment (yoninachmany) [inline web/routes.go]: If so, then we should change line 82's style

File: transcription/transcription.go (hunk 2 above)
Comment (yoninachmany) [inline transcription/transcription.go]: Test this function == mock tests?

not testing `fileNameFromURL`

waiting to test `DownloadFileFromURL` because issues with testing `http.Get`

Comment (sandlerben) [inline transcription/transcription.go]: Yeah, we may use mocking. The issue is that it's not possible to mock functions of the same package (right now) so we can't directly mock `DownloadFileFromUrl`. We would need to mock `http.Get` to test this so that when we call `DownloadFileFromUrl` nothing actually happens.

Can you add that we need to test this function to the issue you created?

Comment (yoninachmany) [inline transcription/transcription.go]: Added to issue https://github.com/hack4impact/audio-transcription-service/issues/27. Is there anything to test now or merge?

File: web/routes.go [hunk 3]
```diff
@@ -64,7 +65,9 @@ func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	fmt.Fprintf(w, "Accepted task "+id+"!")
```
Comment (sandlerben) [inline web/routes.go]: Why `Fprintf` over `w.write`?

Comment (sandlerben) [inline web/routes.go]: If you want to actually use formatting, it should look like

``` go
fmt.Fprintf(w, "Accepted task %d!", id)
```

File: web/routes.go (hunk 1 above)
Comment (sandlerben) [inline web/routes.go]: huh?

File: transcription/transcription.go [hunk 4]
```diff
@@ -7,6 +7,7 @@ import (
 	"os"
 	"os/exec"
 	"strings"
+	// "github.com/hack4impact/audio-transcription-service/web"
```
Comment (sandlerben) [inline transcription/transcription.go]: why is this commented out?

File: web/form.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (sandlerben) [inline web/form.html]: Why is this added here?

Comment (yoninachmany) [inline web/form.html]: this file shouldnt be here...

File: web/routes.go [hunk 6]
```diff
@@ -64,7 +65,10 @@ func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	fmt.Fprintf(w, "Accepted!")
+	executer := tasks.DefaultTaskExecuter
+	id := executer.QueueTask(transcription.MakeTaskFunction(jsonData.AudioURL, jsonData.EmailAddresses))
+
+	fmt.Fprintf(w, "Accepted task %d!", id)
```
Comment (sandlerben) [inline web/routes.go]: Is this for debugging? If this is legitimate logging use the `log` package.

**Comment by sandlerben** [comment]
Looks good besides comments

**Comment by anoshaminai** [comment]
Looks good!
